src/ray/core_worker/store_provider/memory_store/memory_store.cc

```diff
@@ -91,7 +91,7 @@ void GetRequest::Set(const ObjectID &object_id, std::shared_ptr<RayObject> objec
     return;  // We have already hit the number of objects to return limit.
   }
   objects_.emplace(object_id, object);
-  if (objects_.size() == num_objects_) {
+  if (objects_.size() == num_objects_ || object->IsException()) {
```

Comment:
I assume you added this due to a test failure? If so, could you add a C++/Python regression test as well?

Reply:
Yes, I added this because `testGetThrowsQuicklyWhenFoundException ` is broken in direct call mode. In Java, we expect that getting a group of objects should throw the exception quickly if any of the objects is an exception, so we don't need to wait for all objects to be ready. However, with this change, some python test cases are broken. See https://travis-ci.com/ray-project/ray/builds/149270941.

Reply:
I want to confirm that "getting a group of objects should throw the exception quickly if any of the objects is an exception" is the expected behavior, right?

Reply:
That is definitely the expected behavior

Reply:
The failures could be that it's not expected for `wait` to return early if there's an exception and IIRC this code path is used for `wait` as well

Reply:
I'm trying to add a flag to distinguish get and wait. Will add a python test.

Reply:
@edoakes To pass the new python test case, I modified some python code about serialization. Could you take a look?

Reply:
Seems now the failed python test cases in travis are irrelevant.